examples/structured_2d_dgsem/elixir_advection_smview.jl

```diff
@@ -0,0 +1,120 @@
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# Coupled semidiscretization of two linear advection systems using converter functions
+# and mesh views for the semidiscretizations. First we define a parent mesh
+# for the entire physical domain, then we define the two mesh views on the parent.
+#
+# In this elixir, we have a square domain that is divided into a left and right.
+# On each half of the domain, a completely independent SemidiscretizationHyperbolic
+# is created for the linear advection equations. The two systems are coupled in the
+# x-direction.
+# For a high-level overview, see also the figure below:
+#
+# (-1,  1)                                   ( 1,  1)
+#     ┌────────────────────┬────────────────────┐
+#     │    ↑ periodic ↑    │    ↑ periodic ↑    │
+#     │                    │                    │
+#     │     =========      │     =========      │
+#     │     system #1      │     system #2      │
+#     │     =========      │     =========      │
+#     │                    │                    │
+#     │<-- coupled         │<-- coupled         │
+#     │         coupled -->│         coupled -->│
+#     │                    │                    │
+#     │    ↓ periodic ↓    │    ↓ periodic ↓    │
+#     └────────────────────┴────────────────────┘
+# (-1, -1)                                   ( 1, -1)
+
+advection_velocity = (0.2, -0.7)
+equations = LinearScalarAdvectionEquation2D(advection_velocity)
+
+# Create DG solver with polynomial degree = 2 and (local) Lax-Friedrichs/Rusanov flux as surface flux
+solver = DGSEM(polydeg = 3)#, surface_flux = flux_lax_friedrichs)
+
+# Domain size of the parent mesh.
+coordinates_min = (-1.0, -1.0)
+coordinates_max = (1.0, 1.0)
+
+# Cell dimensions of the parent mesh.
+cells_per_dimension = (16, 16)
+
+# Create  drentmesh with 16 x 16 elements
+parent_mesh = StructuredMesh(cells_per_dimension, coordinates_min, coordinates_max)
+
+# Create the two mesh views, each of which takes half of the parent mesh.
+mesh1 = StructuredMeshView(parent_mesh; index_min = (1, 1), index_max = (8, 16))
+mesh2 = StructuredMeshView(parent_mesh; index_min = (9, 1), index_max = (16, 16))
+
+# The coupling function is simply the identity, as we are dealing with two identical systems.
+coupling_function = (x, u, equations_other, equations_own) -> u
+
+# Define the coupled boundary conditions
+boundary_conditions1 = (
+                        # Connect left boundary with right boundary of left mesh
+                        x_neg = BoundaryConditionCoupled(2, (:end, :i_forward), Float64,
+                                                         coupling_function),
+                        x_pos = BoundaryConditionCoupled(2, (:begin, :i_forward), Float64,
```

Comment:
Maybe comment on the usage of the symbols `:end, :i_forward` etc.?

Reply:
Done!